sphinx/themes/basic/static/searchtools.js

```diff
@@ -160,11 +160,22 @@ const Search = {
   _queued_query: null,
   _pulse_status: -1,
 
-  htmlToText: (htmlString) => {
+  htmlToText: (htmlString, anchor) => {
     const htmlElement = new DOMParser().parseFromString(htmlString, 'text/html');
     htmlElement.querySelectorAll(".headerlink").forEach((el) => { el.remove() });
+    if (anchor) {
+      const anchorContent = htmlElement.querySelector(anchor);
```

Comment:
Hi @wlach - apologies for not thinking of this during code review: do you think it would make sense to constrain the anchor query further, to the main document content?

```suggestion
      const anchorContent = htmlElement.querySelector(`[role="main"] ${anchor}`);
```

(I thought of this while writing a [comment](https://github.com/sphinx-doc/sphinx/issues/12052#issuecomment-1986935319) about another issue)

Reply:
Handled by #12105.